### DISCUSSION TITLE
I'm trying to connect python to vtube studio

### DISCUSSION THREAD

**Discussion by HabelKiswanto** [Genteki/pyvts#45] [Q&A]
Hello there, I see that this is a new project hence there is no help in the internet, i would be very much be helped if anyone can actually help me. 

`import asyncio, pyvts
#Trigger
plugin_info = {
    "plugin_name": "trigger hotkey",
    "developer": "OverHome",
    "authentication_token_path": "./pyvts_token.txt"
}

async def main():
    myvts = pyvts.vts(plugin_info=plugin_info)
    await myvts.connect()
    await myvts.request_authenticate_token()  # get token
    await myvts.request_authenticate()  # use token
    response_data = await myvts.request(myvts.vts_request.requestHotKeyList())
    hotkey_list = []
    for hotkey in response_data['data']['availableHotkeys']:
        hotkey_list.append(hotkey['name'])
    send_hotkey_request = myvts.vts_request.requestTriggerHotKey(hotkey_list[0])
    await myvts.request(send_hotkey_request) # send request to play 'My Animation 1'
    await myvts.close()

if __name__ == "__main__":
    asyncio.run(main())`
    
    For the first time its working fine, but vtube studio keep asking for confirmation? Is there a way when as long as the plugin is active, it don't need to keep reconfirming, this is quite important. Thank you very much for your help.
    
   Note: I am making a simple (not so simple) AI that uses openai for brain, elevenlabs for mouth, pyttsx3 for ear, and for the face and features, i really love it if i could use vtube studio.

**Comment by Genteki** [comment]
I think the problem is on
```python
await myvts.request_authenticate_token() # get token
```
Every time you call the function will raise VtubeStudio confirmation.

Just call the `request_authenticate_token()` function once to request and save the token. Then use the token to authenticate with function `request_authenticate()`.

**Reply by HabelKiswanto** [reply]
and how do you use the token?

```
async def main():
    myvts = pyvts.vts(plugin_info=plugin_info)
    await myvts.connect()
    await myvts.request_authenticate()  # use token
    response_data = await myvts.request(myvts.vts_request.requestHotKeyList())
    hotkey_list = []
    for hotkey in response_data['data']['availableHotkeys']:
        hotkey_list.append(hotkey['name'])
    send_hotkey_request = myvts.vts_request.requestTriggerHotKey(hotkey_list[0])
    await myvts.request(send_hotkey_request) # send request to play 'My Animation 1'
    await myvts.close()
```
    
    I removed the reuest authenticcate token but now its keyerror: authenticated

**Reply by HabelKiswanto** [reply]
I don't think the authentication_token_path is working,

**Reply by Genteki** [reply]
```python
import pyvts
import asyncio

async def main():
    myvts = pyvts.vts()
    await myvts.connect()
    try:
        await myvts.read_token()
    except:
        await myvts.request_authenticate_token()
        await myvts.write_token()
    await myvts.request_authenticate()  # use token
    response_data = await myvts.request(myvts.vts_request.requestHotKeyList())
    print(response_data)
    hotkey_list = []
    for hotkey in response_data['data']['availableHotkeys']:
        hotkey_list.append(hotkey['name'])
    send_hotkey_request = myvts.vts_request.requestTriggerHotKey(hotkey_list[0])
    await myvts.request(send_hotkey_request) # send request to play 'My Animation 1'
    await myvts.close()

if __name__ == "__main__":
    asyncio.run(main())
```

Sorry about that, you could use `write_token` manually, I would update the feature later(cuz I am busy with the final exam now😭)
